src/dashboard/components/options/OptionsHome.js

```diff
@@ -35,24 +37,24 @@ export default function OptionsHome(props) {
     }
   }, [preloading, loading])
 
-  // const onChangeBackupLocation = (event) => {
-  //   let file = event.target.files[0]
-  //   if (file) {
-  //     let folder = file.path.split('/')
-  //     let folderPath = folder.slice(0, folder.length - 1).join('/')
-  //     setBackupLocation(folderPath)
-  //     saveSetting('user.backupLocation', folderPath)
-  //   }
-  //   setLoading(false)
-  //   setPreloading(false)
-  // }
+  const onChangeBackupLocation = () => {
+    const title = t('Choose your backup location')
+    const properties = ['openDirectory']
+    const files = dialog.showOpenDialogSync(win, { title, properties })
+    if (files && files.length) {
+      let folderPath = files[0]
+      setBackupLocation(folderPath)
```

Comment:
I'm not sure why we have both state and a setting for the backup location.
Looking at the way that the state for backup location works, we're setting the state inside of the render function(!)
I'm not sure that it's necessary to have state at all here, but if we need it then we should use the `useEffect` hook to trigger changes to it either on the first component mount or when some prop changes.